File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let indices = new Map();
    for(let i=0;i<nums.length;i++) {
        const complement = target - nums[i];
        if(indices.has(complement)) {
            return[indices.get(complement), i];
        }
        
        indices.set(nums[i], i);
    }
    return [];
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        resultList = []
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] != target:
                    continue;
                resultList = [i, j]
        return resultList
                    